apps/mira/components/Swap/Modal/DepositModal/StepRequestDeposit.tsx

```diff
@@ -0,0 +1,164 @@
+import { useConfirmedBlockNumber } from "@/utils/contract";
+import { handleTransactionSuccess } from "@/utils/contract/base";
+import { useRequestDeposit } from "@/utils/contract/CrossbellGateway";
+import { NIL_ADDRESS } from "@/utils/ethers/constants";
+import {
+	getTokenDecimals,
+	parseLog,
+	parseTokenAmount,
+} from "@crossbell/bridge-sdk";
+import { Button, Code, Loader, Space, Text } from "@mantine/core";
+import { BigNumber } from "ethers";
+import { useAtom } from "jotai";
+import { useAccount, useWaitForTransaction } from "wagmi";
+import { useDepositModal } from ".";
+import {
+	formAmountAtom,
+	formMainchainNetworkIdAtom,
+	formSidechainNetworkIdAtom,
+} from "../../store";
+import { requestDepositInfo } from "./store";
+
+export default function StepRequestDeposit() {
+	const [sidechainNetworkId] = useAtom(formSidechainNetworkIdAtom);
+	const [mainchainNetworkId] = useAtom(formMainchainNetworkIdAtom);
+	const [amountStr] = useAtom(formAmountAtom);
+
+	const decimals = getTokenDecimals(sidechainNetworkId, "MIRA");
+	const amount = parseTokenAmount(amountStr, decimals);
+
+	const { address = NIL_ADDRESS } = useAccount();
+
+	const fee = BigNumber.from(0); // TODO:
```

Comment:
todo for what?

Reply:
Seems like a just reserved parameter, but since it's not used in deposit call, it has been removed.

( Code copied from https://github.com/Crossbell-Box/mira/blob/66b2963f9817355da272c4cb6ca2753c2c252278/apps/mira/components/Swap/Modal/WithdrawModal/StepRequestWithdrawal.tsx#L32 )